services/web/server/src/simcore_service_webserver/wallets/_db.py

```diff
@@ -0,0 +1,211 @@
+            wallets.c.owner,
+            wallets.c.thumbnail,
+            wallets.c.status,
+            wallets.c.created,
+            wallets.c.modified,
+        )
+    )
+
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(stmt)
+        output: list[UserWalletGetDB] = []
+        for row in await result.fetchall():
+            output.append(parse_obj_as(UserWalletGetDB, row))
+        return output
+
+
+async def get_wallet_for_user(
+    app: web.Application,
+    user_id: UserID,
+    wallet_id: WalletID,
+) -> UserWalletGetDB:
+    stmt = (
+        select(
+            wallets.c.wallet_id,
+            wallets.c.name,
+            wallets.c.description,
+            wallets.c.owner,
+            wallets.c.thumbnail,
+            wallets.c.status,
+            wallets.c.created,
+            wallets.c.modified,
+            func.max(wallet_to_groups.c.read.cast(INTEGER)).cast(BOOLEAN).label("read"),
+            func.max(wallet_to_groups.c.write.cast(INTEGER))
+            .cast(BOOLEAN)
+            .label("write"),
+            func.max(wallet_to_groups.c.delete.cast(INTEGER))
+            .cast(BOOLEAN)
+            .label("delete"),
+        )
+        .select_from(
+            user_to_groups.join(
+                wallet_to_groups, user_to_groups.c.gid == wallet_to_groups.c.gid
+            ).join(wallets, wallet_to_groups.c.wallet_id == wallets.c.wallet_id)
+        )
+        .where(
+            (user_to_groups.c.uid == user_id)
+            & (user_to_groups.c.access_rights["read"].astext == "true")
+            & (wallets.c.wallet_id == wallet_id)
+        )
+        .group_by(
+            wallets.c.wallet_id,
+            wallets.c.name,
+            wallets.c.description,
+            wallets.c.owner,
+            wallets.c.thumbnail,
+            wallets.c.status,
+            wallets.c.created,
+            wallets.c.modified,
+        )
+    )
+
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(stmt)
+        row = await result.first()
+        if row is None:
+            raise WalletAccessForbiddenError(
+                wallet_id=wallet_id,
+            )
+        return parse_obj_as(UserWalletGetDB, row)
+
+
+async def get_wallet(app: web.Application, wallet_id: WalletID) -> WalletGetDB:
+    stmt = (
+        select(
+            wallets.c.wallet_id,
+            wallets.c.name,
+            wallets.c.description,
+            wallets.c.owner,
+            wallets.c.thumbnail,
+            wallets.c.status,
+            wallets.c.created,
+            wallets.c.modified,
+        )
+        .select_from(wallets)
+        .where(wallets.c.wallet_id == wallet_id)
+    )
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(stmt)
+        row = await result.first()
+        if row is None:
+            raise WalletNotFoundError(wallet_id=wallet_id)
```

Comment:
ok here you do it with the exception. that is what I meant in projects.db.get_wallet_for_project